tests/cmds/test_click_types.py

```diff
@@ -0,0 +1,193 @@
+        self.obj = obj
+
+
+def test_click_tx_fee_type() -> None:
+    # Test uint64 (only used as default)
+    # assert TransactionFeeParamType().convert(uint64(10000), None, None) == uint64(10000)
+
+    # TODO: Test MOJO Logic When Implemented
+
+    # Test Decimal / XCH
+    assert TransactionFeeParamType().convert("0.5", None, None) == uint64(int(0.5 * units["chia"]))
+    assert TransactionFeeParamType().convert("0.000000000001", None, None) == uint64(1)
+    assert TransactionFeeParamType().convert("0", None, None) == uint64(0)
+    # Test Decimal Failures
+    with pytest.raises(BadParameter):
+        TransactionFeeParamType().convert("test", None, None)
+    with pytest.raises(BadParameter):
+        TransactionFeeParamType().convert("0.6", None, None)
+    with pytest.raises(BadParameter):
+        TransactionFeeParamType().convert("0.0000000000001", None, None)  # 0.1 mojos
+    with pytest.raises(BadParameter):
+        TransactionFeeParamType().convert("-0.6", None, None)
+    with pytest.raises(BadParameter):
+        TransactionFeeParamType().convert(overflow_decimal_str, None, None)
+    # Test Type Failures
+    with pytest.raises(BadParameter):
+        TransactionFeeParamType().convert(float(0.01), None, None)
+
+
+def test_click_amount_type() -> None:
+    decimal_cli_amount = CliAmount(mojos=False, amount=Decimal("5.25"))
+    large_decimal_amount = CliAmount(mojos=False, amount=overflow_decimal)
+    mojos_cli_amount = CliAmount(mojos=True, amount=uint64(100000))
+    one_mojo_cli_amount = CliAmount(mojos=False, amount=Decimal("0.000000000001"))
+    # Test CliAmount (Generally is not used)
+    assert AmountParamType().convert(decimal_cli_amount, None, None) == decimal_cli_amount
+
+    # Test uint64 (only usable as default)
+    # assert AmountParamType().convert(uint64(100000), None, None) == mojos_cli_amount
+
+    # TODO: Test MOJO Logic When Implemented
+
+    # Test Decimal / XCH (we don't test overflow because we don't know the conversion ratio yet)
+    assert AmountParamType().convert("5.25", None, None) == decimal_cli_amount
+    assert AmountParamType().convert(overflow_decimal_str, None, None) == large_decimal_amount
+    assert AmountParamType().convert("0.000000000001", None, None) == one_mojo_cli_amount
+    # Test Decimal Failures
+    with pytest.raises(BadParameter):
+        AmountParamType().convert("test", None, None)
+    with pytest.raises(BadParameter):
+        AmountParamType().convert("0.0000000000001", None, None)  # 0.1 mojos
+    with pytest.raises(BadParameter):
+        AmountParamType().convert("-999999", None, None)
+    with pytest.raises(BadParameter):
+        AmountParamType().convert("-0.6", None, None)
+    # Test Type Failures
+    with pytest.raises(BadParameter):
+        AmountParamType().convert(float(0.01), None, None)
+
+    # Test CliAmount Class
+    assert decimal_cli_amount.convert_amount(units["chia"]) == uint64(int(5.25 * units["chia"]))
```

Comment:
left int on outside because sometimes the linters care, but i added the decimal around the string.

Reply:
mypy used to care due to incomplete hinting of our fixed width integers.  I have since improved them (though still incomplete iirc) so it should be ok now.  Unless you have had issues 'recently'?

Reply:
oh ok, ill remove that.